src/rgw/rgw_sal.cc

```diff
@@ -415,4 +415,38 @@ int Object::range_to_ofs(uint64_t obj_size, int64_t &ofs, int64_t &end)
   }
   return 0;
 }
-} // namespace rgw::sal
+
+
+std::string rgw_restore_status_dump(rgw::sal::RGWRestoreStatus status)
```

Comment:
nit: Can we use std:string_view instead? Also pass the argument by reference.

Reply:
@soumyakoduri If I pass agrument by reference, then I won't use the api directly in `formatter->dump_string()`